platform/android/MapboxGLAndroidSDK/src/main/java/com/mapbox/mapboxsdk/maps/IconManager.java

```diff
@@ -126,25 +131,28 @@ void reloadIcons() {
   }
 
   void ensureIconLoaded(Marker marker, MapboxMap mapboxMap) {
-    Icon icon = marker.getIcon();
-    if (icon == null) {
-      icon = IconFactory.getInstance(nativeMapView.getContext()).defaultMarker();
-      marker.setIcon(icon);
-    }
-    if (!icons.contains(icon)) {
-      icons.add(icon);
-      loadIcon(icon);
-    } else {
-      Icon oldIcon = icons.get(icons.indexOf(icon));
-      if (!oldIcon.getBitmap().sameAs(icon.getBitmap())) {
-        throw new IconBitmapChangedException();
+    Context context = nativeMapView.getContext();
+    if (context != null) {
+      Icon icon = marker.getIcon();
+      if (icon == null) {
+        icon = IconFactory.getInstance(context).defaultMarker();
+        marker.setIcon(icon);
+      }
+      if (!icons.contains(icon)) {
+        icons.add(icon);
+        loadIcon(icon);
+      } else {
+        Icon oldIcon = icons.get(icons.indexOf(icon));
+        if (!oldIcon.getBitmap().sameAs(icon.getBitmap())) {
+          throw new IconBitmapChangedException();
+        }
       }
-    }
 
-    // this seems to be a costly operation according to the profiler so I'm trying to save some calls
-    Marker previousMarker = marker.getId() != -1 ? (Marker) mapboxMap.getAnnotation(marker.getId()) : null;
-    if (previousMarker == null || previousMarker.getIcon() == null || previousMarker.getIcon() != marker.getIcon()) {
-      marker.setTopOffsetPixels(getTopOffsetPixelsForIcon(icon));
+      // this seems to be a costly operation according to the profiler so I'm trying to save some calls
+      Marker previousMarker = marker.getId() != -1 ? (Marker) mapboxMap.getAnnotation(marker.getId()) : null;
+      if (previousMarker == null || previousMarker.getIcon() == null || previousMarker.getIcon() != marker.getIcon()) {
```

Comment:
Expression hard to read. What about breaking it down introducing explaining variables?